src/coreclr/debug/daccess/daccess.cpp

```diff
@@ -3124,6 +3125,19 @@ ClrDataAccess::ClrDataAccess(ICorDebugDataTarget * pTarget, ICLRDataTarget * pLe
     m_fEnableDllVerificationAsserts = false;
 #endif
 
+    // TODO: [cdac] Get contract descriptor from exported symbol
+    uint64_t contractDescriptorAddr = 0;
+    //if (TryGetSymbol(m_pTarget, m_globalBase, "DotNetRuntimeContractDescriptor", &contractDescriptorAddr))
+    {
+        m_cdac = CDAC::Create(contractDescriptorAddr, m_pTarget);
```

Comment:
If this is how we are using it, I would avoid the allocation mechanism and instead use move semantics. It will make lifetime much easier to reason about.